core/src/banking_stage.rs

```diff
@@ -584,33 +572,34 @@ impl BankingStage {
     }
 
     fn spawn_vote_worker(
-        id: u32,
-        packet_receiver: BankingPacketReceiver,
+        tpu_receiver: BankingPacketReceiver,
+        gossip_receiver: BankingPacketReceiver,
         decision_maker: DecisionMaker,
         bank_forks: Arc<RwLock<BankForks>>,
         committer: Committer,
         transaction_recorder: TransactionRecorder,
         log_messages_bytes_limit: Option<usize>,
         vote_storage: VoteStorage,
     ) -> JoinHandle<()> {
-        let packet_receiver = PacketReceiver::new(id, packet_receiver);
+        let tpu_receiver = PacketReceiver::new(tpu_receiver);
+        let gossip_receiver = PacketReceiver::new(gossip_receiver);
         let consumer = Consumer::new(
             committer,
             transaction_recorder,
-            QosService::new(id),
+            QosService::new(0),
```

Comment:
would be good to also remove `id` from QosService?

Reply:
That one we cannot because the non-vote threads also have a `QosService`